api/app/controllers/spree/api/v2/storefront/cart_controller.rb

```diff
@@ -114,6 +106,10 @@ def remove_coupon_code
 
           private
 
+          def ensure_order
+            raise ActiveRecord::RecordNotFound if spree_current_order.nil?
```

Comment:
I think we should use a custom error class here. Using the internal Rails error class (`ActiveRecord::RecordNotFound`) may lead to some problems in the future